pkg/pgmodel/model/samples.go

```diff
@@ -12,68 +12,123 @@ import (
 	"github.com/timescale/promscale/pkg/prompb"
 )
 
-type SamplesInfo struct {
-	Series  *Series
-	Samples []prompb.Sample
+type Samples interface {
```

Comment:
Sorry, but I see just one implementation of this. So, is there any other reason for making this interface?

Reply:
Yes in the future we'll plan more (logging, tracing, etc)

Reply:
Then I would say to either add a note there (in case we forget).

Reply:
Considering adding this to the project board would be much suited.